Applied-Research-Winter-2022/Pozter
# Ties together Billboard and Canvas page

## Link Jira ticket:

## Author Notes:

This commit gives an example of:

1. how data fetched from the API can be used to create Vue components that a user can click on
2. how clicking on the Vue component a) sends the user to the next page, and b) includes data that can be accessed in the mounted() hook of the next page

Places where I made changes:

- Billboard page:
  - moved the <v-card> to its own component in src/components/Billboard/BillboardCard.vue
  - in the mounted() hook, an API call (using the DataService) gets all the billboard records in the database and puts them in the billboards array (in the data() object). this billboards array now takes the place of the cards array. A BillboardCard.vue component is created for every item in the billboards array.
- BillboardCard.vue
  - see handleClick method, which pushes the user to the next page **and** packages the id of the billboard they chose in the query object
- Canvas page:
  - mounted() lifecycle hook: the billboard id can be found in the this.$route.query object

## Impacts and Notes:

n/a

## Testing Guidance:

1) Start both the front-end and the server
2) Go to the the billboard page (localhost:8080/create)
3) The cards now correspond to the billboard records in the database. you can now click on one to select a billboard
4) you will be taken to the next page (localhost:8080/create/2). If you look in the console, you should be able to see the id of the billboard you selected on the previous page. We can now use that billboard_id with the DataService to get all the Canvases associated with that billboard!

## Author Checklist:

- [n/a] Change Jira ticket status to into Review.
- [n/a] Updated the documentation.
